### PR TITLE
feat(eslint-plugin): deprecate no-throw-literal and add a renamed only-throw-error

### LINKED ISSUE #6109: [no-throw-literal] Rename to only-throw-error
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/no-throw-literal/

### Description

Per the docs:

> This rule restricts what can be thrown as an exception. When it was first created, it only prevented literals from being thrown (hence the name), but it has now been expanded to only allow expressions which have a possibility of being an Error object.

We've got a v6 breaking change coming up some. Maybe a good time to `replacedBy`-deprecate/rename to a more accurate name?

### Fail

```typescript
n/a
```

### Pass

```typescript
n/a
```

### Additional Info

ESLint core's docs are https://eslint.org/docs/latest/rules/no-throw-literal.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6109
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Effectively renames `no-throw-literal` to `only-throw-error`. This avoids being a breaking change per #6109's suggestion by keeping the rule under the old name as well.

### CODE REVIEW

File: packages/eslint-plugin/docs/rules/no-throw-literal.mdx [hunk 1]
```diff
@@ -12,104 +12,14 @@ import TabItem from '@theme/TabItem';
...
-   */
-  allowThrowingUnknown?: boolean;
-}
-
-const defaultOptions: Options = {
-  allowThrowingAny: false,
-  allowThrowingUnknown: false,
-};
-```
+The current name `no-throw-literal` will be removed in a future major version of typescript-eSLint.
```
Comment (yeonjuan) [inline packages/eslint-plugin/docs/rules/no-throw-literal.mdx:22]:
```suggestion
The current name `no-throw-literal` will be removed in a future major version of typescript-eslint.
```
[Docs] Looks like a typo. :)

File: packages/eslint-plugin/docs/rules/no-throw-literal.mdx [hunk 2]
```diff
@@ -12,104 +12,14 @@ import TabItem from '@theme/TabItem';
 It is considered good practice to only `throw` the `Error` object itself or an object using the `Error` object as base objects for user-defined exceptions.
 The fundamental benefit of `Error` objects is that they automatically keep track of where they were built and originated.
 
-This rule restricts what can be thrown as an exception. When it was first created, it only prevented literals from being thrown (hence the name), but it has now been expanded to only allow expressions which have a possibility of being an `Error` object. With the `allowThrowingAny` and `allowThrowingUnknown`, it can be configured to only allow throwing values which are guaranteed to be an instance of `Error`.
+This rule restricts what can be thrown as an exception.
 
-## Examples
+:::warn
```
Comment (auvred) [inline packages/eslint-plugin/docs/rules/no-throw-literal.mdx:17]:
```suggestion
:::warning
```

It looks like there is no `warn` admonition in Docusaurus :smile: , it should be `warning` or `caution`

https://deploy-preview-8701--typescript-eslint.netlify.app/rules/no-throw-literal

**Review by auvred — CHANGES_REQUESTED** [review]
Looks great overall! But there is small docs issue

File: packages/eslint-plugin/docs/rules/no-throw-literal.mdx (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/docs/rules/no-throw-literal.mdx:17]: D'oh, thanks!

**Pull request merged**